packages/components/src/Alert/src/Alert.tsx

```diff
@@ -167,28 +174,30 @@ function Alert(props:AlertProps, ref: ForwardedRef<HTMLDivElement>) {
                         <ButtonGroup align="end" className={styles["hop-Alert__button-group"]}>
                             {cancelButtonLabel &&
                                 <Button
-                                    onPress={() => chain(renderProps.close(), onCancelButtonClick?.())}
+                                    onPress={() => chain(!isControlled && renderProps.close(), onCancelButtonClick?.())}
                                     variant="secondary"
+                                    isDisabled={isLoading}
                                     autoFocus={autoFocusButton === "cancel"}
                                 >
                                     {cancelButtonLabel}
                                 </Button>
                             }
                             {secondaryButtonLabel &&
                                 <Button
-                                    onPress={() => chain(renderProps.close(), onSecondaryButtonClick?.())}
+                                    onPress={() => chain(!isControlled && renderProps.close(), onSecondaryButtonClick?.())}
                                     variant="secondary"
-                                    isDisabled={secondaryButtonDisabled}
+                                    isDisabled={isLoading || secondaryButtonDisabled}
                                     autoFocus={autoFocusButton === "secondary"}
                                 >
                                     {secondaryButtonLabel}
                                 </Button>
                             }
                             <Button
                                 variant={variant === "confirmation" ? "primary" : "danger"}
+                                isLoading={isLoading}
                                 isDisabled={primaryButtonDisabled}
                                 autoFocus={autoFocusButton === "primary"}
-                                onPress={() => chain(renderProps.close(), onPrimaryButtonClick?.())}
+                                onPress={() => chain(!isControlled && renderProps.close(), onPrimaryButtonClick?.())}
```

Comment:
i don't think this is necessary. if they still have a loading, they can ignore the handleClose event on their side